src/submission/ChildProcess.jl

```diff
@@ -62,21 +67,28 @@ function pigeons(pt_arguments, new_process::ChildProcess)
         run(julia_cmd, wait = new_process.wait)
     else
         mpiexec() do exe
-            mpi_cmd = `$exe -n $(new_process.n_local_mpi_processes)`
-            cmd = `$mpi_cmd $julia_cmd`
+            args    = new_process.mpiexec_args
+            mpi_cmd = length(args)>0 ? `$exe $args` : `$exe` # need this because `$("")` == `''` != `` 
```

Comment:
By making changing `mpiexec_args::String` into`mpiexec_args::Cmd`  you would avoid having to do this. Also, the current implementation would not work if you want to pass in two arguments. Here is some commands to try in the REPL to see why (notice how the REPL underlines the segmentation into the array that will be passed to the child proces):
```
julia> c1 = ``
``

julia> c2 = `test $c1`
`test`

julia> c3 = "asf asd"
"asf asd"

julia> `test $c3`
`test 'asf asd'`

julia> c4 = `asd asdf`
`asd asdf`

julia> `test $c4`
`test asd asdf`
```

Reply:
having weird issues with @kwdef when replacing `::String` with `::Cmd`

```julia
MethodError: Cannot `convert` an object of type String to an object of type Cmd
  Closest candidates are:
    convert(::Type{T}, ::T) where T at Base.jl:61
  Stacktrace:
    [1] ChildProcess(n_threads::Int64, dependencies::Vector{Any}, n_local_mpi_processes::Int64, wait::Bool, mpiexec_args::String)
      @ Pigeons ~/temp/Pigeons.jl/src/submission/ChildProcess.jl:10
    [2] ChildProcess(; n_threads::Int64, dependencies::Vector{Any}, n_local_mpi_processes::Int64, wait::Bool, mpiexec_args::String)
      @ Pigeons ./util.jl:493
```
Note that the last location is wrong: util.jl is part of Base, and the location is inside the definition of @kwdef

Reply:
nevermind, it was a problem on my end that was being masked by kwdef